R/hiper_glm.R

```diff
@@ -1,17 +1,47 @@
 #' @export
-hiper_glm <- function(design, outcome, model="linear") {
+hiper_glm <- function(design, outcome, model="linear", option = list()) {
 
   supported_model <- c("linear", "logit")
 
   if (!(model %in% supported_model)) {
     stop(sprintf("The model %s is not supported.", model))
   }
 
-  warning("`hiper_glm` is yet to be implemented.")
+  coef_estimate <- NULL
+
+  if (!is.null(option[["mle_solver"]])) {
+
+    if (option[["mle_solver"]] == "BFGS") {
+      # message("Using BFGS\n")
+
+      op <- stats::optim(par = rnorm(dim(design)[2]),
```

Comment:
-2

Break the BFGS solverout into its own function, e.g.

```
 if(option["mle_solver"] == "BFGS"){
    model_coefs = mle_bfgs(...)
    } else {
    model_coefs = mle_pinv(...)
    }
```

Functions should be simple and only do one thing. In this case hiper_glm is just orchestrating the right function to call, and if you change how pinv works (e.g. going from cholesky to LU), hiper_glm never has to know (and shouldn't have to know)